endpoints/openrtb2/amp_auction.go

```diff
@@ -578,21 +588,46 @@ func setAmpExt(site *openrtb2.Site, value string) {
 	}
 }
 
-func readPolicy(consent string) (privacy.PolicyWriter, error) {
-	if len(consent) == 0 {
+// setRegExtGDPR sets regs.ext.gdpr to either 0 or 1 only if it was set in the request query
+func setRegExtGDPR(gdprApplies *bool, req *openrtb2.BidRequest) error {
+	if gdprApplies == nil {
+		return nil
+	}
+
+	gdpr := int8(0)
+	if *gdprApplies {
+		gdpr++
+	}
+
+	reqWrap := &openrtb_ext.RequestWrapper{BidRequest: req}
+	if regsExt, err := reqWrap.GetRegExt(); err == nil {
+		regsExt.SetGDPR(&gdpr)
+	} else {
+		return err
+	}
+	return reqWrap.RebuildRequest()
+}
+
+func (pr policyReader) readPolicy(ampParams amp.Params, req *openrtb2.BidRequest) (privacy.PolicyWriter, error) {
+
+	if err := setRegExtGDPR(ampParams.GdprApplies, req); err != nil {
```

Comment:
I'd like to see writing the GDPR flag incorporated into the consentwriter. This will help to ensure we do not write the gdpr flag unless the we know this is a TCF1 or TCF2 consent type.

Reply:
That's the reason I wanted to bring this changes up for discussion in a PR. According to my interpretation of the specs, I understood this to be independent. Paraphrasing the relevant section as described in issue  #1817: `"2. If gdpr_applies is supplied, use its value to set regs.ext.gdpr: gdpr_applies=true means regs.ext.gdpr:1, gdpr_applies=false means regs.ext.gdpr:0, any other value means regs.ext.gdpr is not set."`. This seems to be TCF1 and TCF2 independent. Am I wrong?

Reply:
It might be... I'm really not sure.

Reply:
> I'd like to see writing the GDPR flag incorporated into the consentwriter. This will help to ensure we do not write the gdpr flag unless the we know this is a TCF1 or TCF2 consent type.
The GDPR flag is independent of the consent according to issue #1817. Let me know if you agree with the approach of keeping the `setRegExtGDPR()` in the `amp_auction.go` endpoint as part of the request parsing process.